hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRegionLocatorHelper.java

```diff
@@ -50,10 +49,17 @@ static boolean canUpdateOnError(HRegionLocation loc, HRegionLocation oldLoc) {
       oldLoc.getServerName().equals(loc.getServerName());
   }
 
+  static void updateCachedLocationOnError(HRegionLocation loc, Throwable exception,
+      Function<HRegionLocation, HRegionLocation> cachedLocationSupplier,
+      Consumer<HRegionLocation> addToCache, Consumer<HRegionLocation> removeFromCache) {
+    updateCachedLocationOnError(loc, exception, cachedLocationSupplier, addToCache,
+      removeFromCache, null);
+  }
+
   static void updateCachedLocationOnError(HRegionLocation loc, Throwable exception,
       Function<HRegionLocation, HRegionLocation> cachedLocationSupplier,
       Consumer<HRegionLocation> addToCache, Consumer<HRegionLocation> removeFromCache,
-      Optional<MetricsConnection> metrics) {
+      MetricsConnection metrics) {
```

Comment:
If the metrics could be null then I do not think it is necessary to provide the method above? Just pass null when calling.

Reply:
Done